Rarity badges instead of rarity stars

mfkaptan opened the pull request:
- Use rectangular badges
- Red, orange and green for rare, uncommon and common respectively
- Update `poke-type` to take a single element instead of an array
- Fix type badge issues
- Fix #127 

issue #69 
can be used for #121

mfkaptan committed:
Replace rarity stars with rarity badges

mfkaptan committed:
Merge branch 'develop' into pokedex-index-improvement

mfkaptan committed:
Use rectangular color badges for rarity
Use red, orange and green for rare, uncommon and common respectively
issue #69

mfkaptan commented:
Thanks for the review @WoH - the night coder!

sacdallago commented:
![](https://s-media-cache-ak0.pinimg.com/236x/ec/f3/c0/ecf3c042507787ed25ed2f57b8e3aa94.jpg)

MajorBreakfast commented:
I think you should use a simple div instead of an `<ion-badge>`. The whole point of `<ion-badge>` is to get an appearance native to the operating system. That's about the opposite what we're using it for.

mfkaptan committed:
Merge branch 'develop' into pokedex-index-improvement

mfkaptan committed:
Fix badge issues
- Move rarity badge next to pokemon id
- Use simple div instead of ion badge for rarity
- Add margin to type badges
issue #69

mfkaptan committed:
Update `type` to be plural"

mfkaptan commented:
How it looks like with the new changes:

![image](https://cloud.githubusercontent.com/assets/3976573/19389731/56ed7880-9225-11e6-9abb-5a615fdf051e.png)

![image](https://cloud.githubusercontent.com/assets/3976573/19389741/67eebde2-9225-11e6-9782-f885144e783b.png)

![image](https://cloud.githubusercontent.com/assets/3976573/19390196/502e0706-9227-11e6-9000-80598623db2e.png)

MajorBreakfast commented on ionic2/app/components/rarity-badge/rarity-badge.component.scss:
```diff
@@ -1,15 +1,16 @@
-.common {
-  background-color: green;
+div {
+  border-radius: 0px;
```
Why?

MajorBreakfast commented on ionic2/app/components/types/types.component.html:
```diff
@@ -1,3 +1,3 @@
-<ion-badge [style.background-color]="poketype.color" *ngFor="let poketype of typeArray">
+<ion-badge [style.background-color]="poketype.color" *ngFor="let poketype of typesArray">
   {{ poketype.element }}
 </ion-badge>
```
I think you should make that a plain div as well

MajorBreakfast commented on ionic2/app/components/types/types.component.html:
```diff
@@ -1,3 +1,3 @@
-<ion-badge [style.background-color]="poketype.color" *ngFor="let poketype of typeArray">
+<ion-badge [style.background-color]="poketype.color" *ngFor="let poketype of typesArray">
```
Don't do the ngFor in the component, do it on the page

mfkaptan replied:
May I ask what is the reason behind this? We have to change this in different places too. Couldn't it just stay like this?

MajorBreakfast commented on ionic2/app/components/types/types.component.scss:
```diff
@@ -1,4 +1,5 @@
 ion-badge{
```
missing space

MajorBreakfast commented on ionic2/app/components/types/types.component.scss:
```diff
@@ -1,4 +1,5 @@
 ion-badge{
   margin-left: 2.5px;
   margin-right: 2.5px;
+  margin-bottom: 2.5px;
```
margin combo property might be better

MajorBreakfast commented on ionic2/app/components/types/types.component.ts:
```diff
@@ -9,17 +9,17 @@ import { ApiService } from '../../services/api.service';
-  @Input() type: string[];
-  typeArray: Object[];
+  @Input() types: string[];
+  typesArray: Object[];
```
In JavaScript it's uncustomary to end arrays with "-Array". Just use the plural

MajorBreakfast commented on ionic2/app/components/types/types.component.ts:
```diff
@@ -9,17 +9,17 @@ import { ApiService } from '../../services/api.service';
-      this.typeArray.push({element:item, color:typeColors[item]})
+    this.typesArray = [];
+    for (let item of this.types) {
+      this.typesArray.push({element:item, color:typeColors[item]})
```
Don't modify it, make a new property

MajorBreakfast replied:
You should take a look at `array.map(function (value) { return .... })`

MajorBreakfast commented on ionic2/app/pages/poke-detail/poke-detail.page.html:
```diff
@@ -21,11 +25,8 @@
-            <poke-rarity-badge [rarity]="pokemon.appearanceLikelihood"></poke-rarity-badge>
-            <span style="opacity: 0.5">(Rarity: Soon badge instead of stars)</span>
-          </div-->
+          <div id="types"><poke-type [types]="pokemon.types"></poke-type></div>
```
Still bad. The component should be called `<poke-type>` it should have a `type` property and it should accept a single value (not an array)

mfkaptan replied:
Can we just change it to poke-types?

mfkaptan committed:
Update `poke-type` to take a single element
instead of an array
- Replace ion-badge with div for type badges
- Increase border-radius of type badges
- fix #127
issue #69

mfkaptan committed:
Use light ion badge for id

MajorBreakfast commented:
The source code looks good. Feel feel to merge if everything works. Good work!
